examples/platform/silabs/provision/ProvisionStorageDefault.cpp

```diff
@@ -70,16 +70,26 @@ CHIP_ERROR ErasePage(uint32_t addr)
     return chip::DeviceLayer::Silabs::GetPlatform().FlashErasePage(addr);
 }
 
-CHIP_ERROR WritePage(uint32_t addr, const uint8_t * data, size_t size)
+size_t RoundNearest(size_t n, size_t multiple)
 {
-    return chip::DeviceLayer::Silabs::GetPlatform().FlashWritePage(addr, data, size);
+    return (n % multiple) > 0 ? n + (multiple - n % multiple) : n;
 }
 
-size_t RoundNearest(size_t n, size_t multiple)
+CHIP_ERROR WritePage(uint32_t addr, const uint8_t * data, size_t size, size_t max)
 {
-    return (n % multiple) > 0 ? n + (multiple - n % multiple) : n;
+    size_t size_32 = RoundNearest(size, 4);
```

Comment:
This implies that this is the size rounded in _uint32_, while the name suggest that we are writing a whole page.

My comment below might be wrong because I see that the logic doesn't seem to indicate that `multiple` matches the page size... which is why this is in dire need of commenting.

Reply:
You were not far-off. The size is not rounded to the page size, but the nearest 32-bit boundary. Otherwise the flash driver fails to write. I have updated the comment accodingly.